Feature/rdkoss 217

maniselva006c opened the pull request.

Commit:
RDKOSS-260: Enable uninative support

Reason for the change: It helps to isolate the build system from the
host distribution’s C library in order to make re-use of native shared
state artifacts across different host distributions.

Signed-off-by: mselva006c <mani_selvaraj@comcast.com>

Commit:
RDKOSS-261: Create a build target for native packages

Reason for the change: The RDK OSS native packages assist in building the necessary native packages for RDK builds. This target can be utilized to create a locked sstate with native packages and toolchain.

Signed-off-by: mselva006c <mani_selvaraj@comcast.com>

Commit:
RDKOSS-267: Include locked sstate from host

Reason for the change: Utilize the locked sstate from the host machine or Docker if configured via HOST_LOCKED_SSTATE_PATH

Signed-off-by: mselva006c <mani_selvaraj@comcast.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/meta-rdk-oss-reference?pullRequest=78) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/meta-rdk-oss-reference?pullRequest=78) before we can accept your contribution.<br/><hr/>**mselva006c** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/meta-rdk-oss-reference?pullRequest=78) it.</sub>

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 0 files pending approval, 1 file pending identification.
* Protex Server Path: /home/blackduck/github/meta-rdk-oss-reference/78/rdkcentral/meta-rdk-oss-reference

* Commit: 9977de5c0b116006182c9d68055e056e513b0377

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/ee243c051fab49a4c5aa41c1f375c6dc)'

rdkcmf-jenkins commented:
b'## **WARNING:** A Blackduck scan failure has been waived

A prior failure has been upvoted
* Upvote reason: OK

* Commit: 9977de5c0b116006182c9d68055e056e513b0377
'

Commit:
RDKOSS-283: Include addtional native packages

Reason for the change: Include addtional native packages to the rdk-oss-nativepackages target

Signed-off-by: mselva006c <mani_selvaraj@comcast.com>

Commit:
RDKOSS-287: Add core OSS packages for toolchain construction

Reason for the change: Include core oss packages built with the target architecture necessary for constructing the fundamental toolchain.

Signed-off-by: mselva006c <mani_selvaraj@comcast.com>

Commit:
RDKOSS-299: Enable locked sstate support based on distro feature

Reason for change: Include the signature file based on tune profile and distro feature

Signed-off-by: mselva006c <mani_selvaraj@comcast.com>

Commit:
RDKOSS-295: Fix retriggering of "do_fetch" task for native modules

Reason for change: Previously, the "do_fetch" task for all native modules was retriggered for the image target, as opposed to other tasks that were sourced from the locked sstate cache. This issue arose due to the addition of the "do_populate_lic" task to all native recipes for the image target. This commit addresses the problem, ensuring consistent retrieval of tasks from the sstate cache.

Signed-off-by: mselva006c <mani_selvaraj@comcast.com>

Commit:
RDKOSS-300: Include chrpath-replacement-native

Reason for the change: Include chrpath-replacement-native to the rdk-oss-nativepackages package

Signed-off-by: mselva006c <mani_selvaraj@comcast.com>
